integration-tests/src/test/java/oracle/kubernetes/operator/ItModelInImageOverride.java

```diff
@@ -0,0 +1,246 @@
+    // params passed to mii
+    domainMap.put("wdtModelFile", "./model.wls.yaml");
+    domainMap.put("wdtModelPropertiesFile", "./model.empty.properties");
+    // params to create cm
+    String cmModelFile = "./model.properties";
+    domainMap.put("miiConfigMap", cmName);
+    domainMap.put("miiConfigMapFileOrDir", cmModelFile);
+
+    // create domain and verify
+    domain = TestUtils.createDomain(domainMap);
+    domain.verifyDomainCreated();
+  }
+
+  private void wdtConfigOverride() throws Exception {
+    LoggerHelper.getLocal().log(Level.INFO, "Creating configMap");
+    String origDir = BaseTest.getProjectRoot()
+        + "/integration-tests/src/test/resources/model-in-image";
+    String origModelFile = origDir + "/model.jdbc.yaml";
+    String origPropFile = origDir + "/model.jdbc.properties";
+    String destDir = getResultDir() + "/samples/model-in-image-override";;
+    String destModelFile = destDir + "/model.jdbc_2.yaml";
+    String destPropFile = destDir + "/model.jdbc_2.properties";
+    Files.createDirectories(Paths.get(destDir));
+
+    TestUtils.copyFile(origModelFile, destModelFile);
+    TestUtils.copyFile(origPropFile, destPropFile);
+
+    // Re-create config map after deploying domain crd
+    final String domainUid = domain.getDomainUid();
+    final String cmName = domainUid + configMapSuffix;
+    final String label = "weblogic.domainUID=" + domainUid;
+
+    TestUtils.createConfigMap(cmName, destDir, domainNS, label);
+  }
+
+  private void modifyDomainYamlWithRestartVersion()
+      throws Exception {
+    String originalYaml =
+        getUserProjectsDir()
+            + "/weblogic-domains/"
+            + domain.getDomainUid()
+            + "/domain.yaml";
+
+    // Modify the original domain yaml to include restartVersion in admin server node
+    DomainCrd crd = new DomainCrd(originalYaml);
+    Map<String, String> objectNode = new HashMap();
+    objectNode.put("restartVersion", "v1.1");
+    crd.addObjectNodeToDomain(objectNode);
+    String modYaml = crd.getYamlTree();
+    LoggerHelper.getLocal().log(Level.INFO, modYaml);
+
+    // Write the modified yaml to a new file
+    Path path = Paths.get(getUserProjectsDir()
+        + "/weblogic-domains/"
+        + domain.getDomainUid(), "modified.domain.yaml");
+    LoggerHelper.getLocal().log(Level.INFO, "Path of the modified domain.yaml :{0}", path.toString());
+    Charset charset = StandardCharsets.UTF_8;
+    Files.write(path, modYaml.getBytes(charset));
+
+    // Apply the new yaml to update the domain crd
+    LoggerHelper.getLocal().log(Level.INFO, "kubectl apply -f {0}", path.toString());
+    ExecResult exec = TestUtils.exec("kubectl apply -f " + path.toString());
+    LoggerHelper.getLocal().log(Level.INFO, exec.stdout());
+    LoggerHelper.getLocal().log(Level.INFO, "Verifying if the domain is restarted");
+    domain.verifyDomainRestarted();
+  }
+
+  private void verifyJdbcOverride() throws Exception {
+    // get domain name
```

Comment:
this is a really nice method, logs what it is doing, puts the command in a string, which it prints and execs, so no chance of someone coming along later and updating one but not the other, the assertion is specific and has a reasonable message.

Reply:
Thanks,Mark for encouragement :)